source/assets/js/playground/autocomplete.ts

```diff
@@ -0,0 +1,245 @@
+      acc[mod.name] =
+        mod.variables?.map(variable => ({
+          label: `${mod.name}.${variable}`,
+          type: 'variable',
+        })) || [];
+      return acc;
+    },
+    {}
+  )
+);
+
+// Completion results for module functions.
+const moduleFunctionsCompletions = Object.freeze(
+  moduleMetadata.reduce(
+    (acc: {[k: string]: CompletionResult['options'] | []}, mod) => {
+      acc[mod.name] =
+        mod.functions.map(func => ({
+          label: `${mod.name}.${func}`,
+          apply: `${mod.name}.${func}(`,
+          type: 'method',
+          boost: 10,
+          validFor: identifier,
+        })) || [];
+      return acc;
+    },
+    {}
+  )
+);
+
+// Type predicate for modules names.
+function isModuleName(string?: string | null): string is ModuleName {
+  return moduleNames.includes(string as ModuleName);
+}
+
+// Returns the list of built in modules that are included in the text.
+function includedmoduleMetadata(state: EditorState): ModuleName[] {
+  const tree = syntaxTree(state);
+  const useNodes = tree.topNode.getChildren('UseStatement');
+  const usedModules = useNodes.map(useNode => {
+    const cursor = useNode.cursor();
+    while (cursor.next()) {
+      if (cursor.node.name === 'StringLiteral') {
+        const string = state.doc.sliceString(cursor.from, cursor.to);
+        return string.match(moduleUseRegex)?.groups?.modName;
+      }
+    }
+    return null;
+  });
+
+  return usedModules.filter<ModuleName>(mod => isModuleName(mod));
+}
+
+// Completions for the namespaces of included built in modules.
+function builtinModuleNameCompletion(
+  context: CompletionContext
+): CompletionResult | null {
+  const nodeBefore = syntaxTree(context.state).resolveInner(context.pos, -1);
+  if (nodeBefore.type.name !== 'ValueName') return null;
+  // Prevent module name from showing up after `.`
+  if (nodeBefore.parent?.type.name === 'NamespacedValue') return null;
+  const includedModules = includedmoduleMetadata(context.state);
+
+  const match = context.matchBefore(/\w+/);
+  if (!match) return null;
+
+  return {
+    from: match.from,
+    to: match.to,
+    options: includedModules.map(mod => ({
+      label: mod,
+      info: moduleMetadata.find(builtin => builtin.name === mod)?.description,
+      type: 'namespace',
+      boost: 20,
+      validFor: identifier,
+    })),
+  };
+}
+
+// Completions for variables and functions for included built in modules.
+function builtinModuleItemCompletion(
+  context: CompletionContext
+): CompletionResult | null {
+  const nodeBefore = syntaxTree(context.state).resolveInner(context.pos, -1);
+  if (
+    ![nodeBefore.type.name, nodeBefore.parent?.type.name].includes(
+      'NamespacedValue'
+    )
+  ) {
+    return null;
+  }
+  const moduleNameMatch = context.matchBefore(moduleNameRegExp);
+
+  if (!moduleNameMatch) return null;
+  if (moduleNameMatch.from === moduleNameMatch.to && !context.explicit)
+    return null;
```

Comment:
Brackets

Reply:
Fixed in https://github.com/sass/sass-site/pull/1166/commits/7c4ef88299d155c3d4db97b36d502220278162a6